### PR TITLE
Add prometheus support for k8s 1.25.x

### PR DESCRIPTION
**Pull request by laverya**
#### What this PR does / why we need it:
This adds a new version of Prometheus, and updates the Prometheus values.yaml to prevent the inclusion of PodSecurityPolicies allowing it to be installed on k8s 1.25.x clusters.

Closes https://github.com/replicatedhq/kURL/pull/3454.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
  Adds [Prometheus add-on](https://kurl.sh/docs/add-ons/prometheus) version 0.59.1-40.1.0.
```

### CODE REVIEW

**Comment by laverya** [comment]
And now the error is:
```
error: error validating "./kustomize/prometheus/operator/": error validating data: ValidationError(Deployment.spec.template.spec.containers[0].securityContext): unknown field "resources" in io.k8s.api.core.v1.SecurityContext; if you choose to ignore these errors, turn validation off with --validate=false
```

**Comment by laverya** [comment]
Oh shoot, apparently the default set of labelselectors changed
so I'm going to need to remove the current ones and add a new override
Well, we've already got a pattern for it so it won't be hard


```
The DaemonSet "prometheus-node-exporter" is invalid: spec.selector: Invalid value: v1.LabelSelector{MatchLabels:map[string]string{"app.kubernetes.io/name":"prometheus-node-exporter"}, MatchExpressions:[]v1.LabelSelectorRequirement(nil)}: field is immutable
```

**Comment by laverya** [comment]
Manifest diff:
```
diff 0.58.0-39.12.1 0.59.1-40.1.0
diff 0.58.0-39.12.1/Manifest 0.59.1-40.1.0/Manifest
4c4
< image kube-state-metrics registry.k8s.io/kube-state-metrics/kube-state-metrics:v2.5.0
---
> image kube-state-metrics registry.k8s.io/kube-state-metrics/kube-state-metrics:v2.6.0
6c6
< image prometheus quay.io/prometheus/prometheus:v2.37.0
---
> image prometheus quay.io/prometheus/prometheus:v2.38.0
8,9c8,9
< image prometheus-config-reloader quay.io/prometheus-operator/prometheus-config-reloader:v0.58.0
< image prometheus-operator quay.io/prometheus-operator/prometheus-operator:v0.58.0
---
> image prometheus-config-reloader quay.io/prometheus-operator/prometheus-config-reloader:v0.59.1
> image prometheus-operator quay.io/prometheus-operator/prometheus-operator:v0.59.1
Common subdirectories: 0.58.0-39.12.1/crds and 0.59.1-40.1.0/crds
diff 0.58.0-39.12.1/install.sh 0.59.1-40.1.0/install.sh
4c4
<     local src="$DIR/addons/prometheus/0.58.0-39.12.1"
---
>     local src="$DIR/addons/prometheus/0.59.1-40.1.0"
Common subdirectories: 0.58.0-39.12.1/operator and 0.59.1-40.1.0/operator
```

File: addons/prometheus/0.59.1-40.1.0/install.sh [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+    GRAFANA_ADMIN_USER=admin
+    GRAFANA_ADMIN_PASS=$(< /dev/urandom tr -dc A-Za-z0-9 | head -c9)
+
+    insert_resources "$grafanadst/kustomization.yaml" grafana-secret.yaml
+
+    render_yaml_file "$src/tmpl-grafana-secret.yaml" > "$grafanadst/grafana-secret.yaml"
+}
+
+function prometheus_outro() {
```
Comment (ricardomaraschini) [inline addons/prometheus/0.59.1-40.1.0/install.sh:97]: Could not find where this is used.

File: addons/prometheus/0.59.1-40.1.0/operator/matchlabel-release.yaml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+- op: remove
+  path: /spec/selector/matchLabels/release
```
Comment (ricardomaraschini) [inline addons/prometheus/0.59.1-40.1.0/operator/matchlabel-release.yaml:2]: For my knowledge, why don't we yank this selector directly from inside `default.yaml` ?

File: addons/prometheus/0.59.1-40.1.0/install.sh (hunk 1 above)
Comment (laverya) [reply addons/prometheus/0.59.1-40.1.0/install.sh:97]: `ADDONNAME_outro` functions are called by the install loop

https://github.com/replicatedhq/kURL/blob/e1436961009c1a0d7bb31e7ec23c19641ed1573c/scripts/common/addon.sh#L187-L191

File: addons/prometheus/0.59.1-40.1.0/operator/matchlabel-release.yaml (hunk 2 above)
Comment (laverya) [reply addons/prometheus/0.59.1-40.1.0/operator/matchlabel-release.yaml:2]: What do you mean?

File: addons/prometheus/template/base/operator/matchlabel-instance.yaml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+- op: remove
+  path: /spec/selector/matchLabels/app.kubernetes.io~1instance
+- op: remove
+  path: /spec/selector/matchLabels/app.kubernetes.io~1name
```
Comment (ricardomaraschini) [inline addons/prometheus/template/base/operator/matchlabel-instance.yaml:4]: is this supposed to be like that ?  I mean, wasn't it supposed to be `app.kubernetes.io/instance` and `app.kubernetes.io/name` ?

Comment (laverya) [reply addons/prometheus/template/base/operator/matchlabel-instance.yaml:4]: You can't have a `/` in the path - you need to escape it. `~1` is escaped `/`, `~0` is escaped `~`.

Comment (laverya) [reply addons/prometheus/template/base/operator/matchlabel-instance.yaml:4]: From jsonpatch.com,

```
If you need to refer to a key with ~ or / in its name, you must escape the characters with ~0 and ~1 respectively. For example, to get "baz" from { "foo/bar~": "baz" } you’d use the pointer /foo~1bar~0.
```

**Comment by emosbaugh** [comment]
> e-state-metrics registry.k8s.io/kube-state-metrics/kube-state-metrics:v2.5.0
> ---
> > image kube-state-metrics registry.k8s.io/kube-state-metrics/kube-state-metrics:v2.6.0
> 6c6
> < image prometheus quay.io/prometheus/prometheus:v2.37.0
> ---
> > image prometheus quay.io/prometheus/prometheus:v2.38.0
> 8,9c8,9
> < image prometheus-config-reloader quay.io/prometheus-operator/prometheus-config-reloader:v0.58.0
> < image prometheus-operator quay.io/prometheus-operator/prometheus-operator:v0.58.0
> ---
> > image prometheus-config-reloader quay.io/prometheus-operator/prometheus-config-reloader:v0.59.1
> > image prometheus-operator quay.io/prometheus-operator/prometheus-operator:v0.59.1
> Common subdirectories: 0.58.0-39.12.1/crds and 0.59.1-40.1.0/crds
> diff 0.58.0-39.12.1/install.sh 0.59.1-40.1.0/install.sh
> 4c4
> <     local src="$DIR/addons/prometheus/0.58.0-39.12.1"
> ---
> >     local src="$DIR/addons/prometheus/0.59.1-40.1.0"
> Common subdirectories: 0.58.0-39.12.1/operator and 0.59.1-40.1.0/operator

I would expect to see the PSPs removed

**Comment by laverya** [comment]
> I would expect to see the PSPs removed

That's just the diff of the files in the root (it's non-recursive) - the full diff is much larger and includes a few CRD changes

File: addons/prometheus/0.59.1-40.1.0/operator/matchlabel-release.yaml (hunk 2 above)
Comment (ricardomaraschini) [reply addons/prometheus/0.59.1-40.1.0/operator/matchlabel-release.yaml:2]: Yeah, that was my question :-D Let's resolve this comment.

File: addons/prometheus/template/base/operator/matchlabel-instance.yaml (hunk 3 above)
Comment (ricardomaraschini) [reply addons/prometheus/template/base/operator/matchlabel-instance.yaml:4]: Very intuitive. OK, lesson learned, thanks for clarifying.

File: addons/prometheus/0.59.1-40.1.0/install.sh (hunk 1 above)
Comment (ricardomaraschini) [reply addons/prometheus/0.59.1-40.1.0/install.sh:97]: Good!

**Review by ricardomaraschini — APPROVED** [review]

**Pull request merged**